lib/io/hardware_i2c.cpp

```diff
@@ -69,11 +69,31 @@ std::optional<std::uint8_t> HardwareI2c::readByte(const std::uint8_t device_addr
 core::Result HardwareI2c::writeByteToRegister(const std::uint8_t device_address,
                                               const std::uint8_t register_address,
                                               const std::uint8_t data)
+{
+  const std::uint8_t register_address_array[2] = {0x00, register_address};
+  return writeByteToDevice(device_address, register_address_array, data);
+}
+
+core::Result HardwareI2c::writeByteToRegister(const std::uint8_t device_address,
+                                              const std::uint16_t register_address,
+                                              const std::uint8_t data)
+{
+  // TODOLater - Is this the best way to do this?
+  const std::uint8_t register_address_hi       = register_address >> 8;
+  const std::uint8_t register_address_lo       = static_cast<std::uint8_t>(register_address);
+  const std::uint8_t register_address_array[2] = {register_address_hi, register_address_lo};
+  return writeByteToDevice(device_address, register_address_array, data);
+}
+
+// TODOLater - Test code
+core::Result HardwareI2c::writeByteToDevice(const std::uint8_t device_address,
+                                            const std::uint8_t register_address_array[2],
+                                            const std::uint8_t data)
```

Comment:
make this more generic, type signature should be 
```suggestion
core::Result HardwareI2c::writeBytesToDevice(const std::uint8_t device_address,
                                            const std::vector<std::uint8t> bytes)
```
and then alter the write to be able to deal with any length of array